api/api_ethereum.go

```diff
@@ -502,52 +503,218 @@ type EthRPCTransaction struct {
 	S                *hexutil.Big    `json:"s"`
 }
 
+// newEthRPCTransactionFromBlockAndIndex creates an EthRPCTransaction from block and index parameters.
+func newEthRPCTransactionFromBlockAndIndex(b *types.Block, index uint64) (*EthRPCTransaction, error) {
+	txs := b.Transactions()
+	if index >= uint64(len(txs)) {
+		return nil, errors.New("invalid transaction index")
+	}
+	return newEthRPCTransaction(txs[index], b.Hash(), b.NumberU64(), index)
+}
+
+// newEthRPCTransaction creates an EthRPCTransaction from Klaytn transaction.
+func newEthRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber, index uint64) (*EthRPCTransaction, error) {
+	// When an unknown transaction is requested through rpc call,
+	// nil is returned by Klaytn API, and it is handled.
+	if tx == nil {
+		return nil, nil
+	}
+
+	from := getFrom(tx)
+
+	// If to is nil, it is fills with from.
+	to := tx.To()
+	if to == nil {
+		to = &from
+	}
```

Comment:
Contract deploying transaction has `nil` for `to` address. This logic should be applied only for few Klaytn tx types.

Reply:
I thought when result of the `tx.To()` is `nil`, then fill the to field to from can cover only few Klaytn tx types. Is there any worry about this?

Reply:
I think when the transaction type is `SmartContractDeploy`, we have to leave `nil` as it is.
So additional tx type checking logic should be appended.
(**should consider FeeDelegation and PartialFeeDelegation also**)

* When the type of Tx is `SmartContractDeploy` transaction. | => Use `nil` value as it is.
* When the type of Tx is `AccountUpdate` transaction . | => Use `from` value instead of `nil`.
* When the type of Tx is `Cancel` transaction. | => Use `from` value instead of `nil`.
* When the type of Tx is `ChainDataAnchoring` transaction. | => Use `from` value instead of `nil`.